cli/cmd/main.go

```diff
@@ -895,27 +889,26 @@ func createDataFilter(c *cli.Context) (*datapb.Filter, error) {
 	if len(c.StringSlice(dataFlagMimeTypes)) != 0 {
 		filter.MimeType = c.StringSlice(dataFlagMimeTypes)
 	}
-	if c.StringSlice(dataFlagTags) != nil && c.String(dataFlagTaggedUntagged) != "" {
-		return nil, errors.New("cannot filter based on matching and tagged or untagged")
-	}
 	if c.StringSlice(dataFlagTags) != nil {
-		filter.TagsFilter = &datapb.TagsFilter{
-			Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_MATCH_BY_OR,
-			Tags: c.StringSlice(dataFlagTags),
-		}
-	}
-	if c.String(dataFlagTaggedUntagged) != "" {
-		switch c.String(dataFlagTaggedUntagged) {
+		tagsLen := len(c.StringSlice(dataFlagTags))
+		switch c.StringSlice(dataFlagTags)[0] {
```

Comment:
Does this NPE if passed an empty list (if that's possible through the cli even)? If so, should probably add a len==0 check first?

Reply:
`StringSlice()` returns nil if not set, but imo for readability/usability, this and the above two checks can be if `len(c.StringSlice(dataFlagTags)) > 0`

In this case, makes it clear that it's valid to grab from the first index

Reply:
What if it's set but empty? Like `org_ids=a,b --tags= robot_name=name`. Or is that just interpreted as not being set?

Reply:
Just ran it locally, it's interpreted as an array of len 1 with an empty string.